### PR TITLE
nojira | @rebeccahongsf | add slideshow wrapper component

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- add slideshow wrapper component

# Review By (Date)
- When possible

### CODE REVIEW

File: src/components/elements/slideshow.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          slidesToScroll: 1,
+          centerMode: false,
+        },
+      },
+    ],
+  };
+  return (
+    <div className="relative w-full">
+      <Slider ref={arrowRef} {...settings}>
+        {children}
```
Comment (rebeccahongsf) [inline src/components/elements/slideshow.tsx:71]: Hmm there's a type issue here. I'll look into it 👀

File: src/components/elements/slideshow.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+'use client';
+import React, { useRef } from 'react';
+import Slider, { CustomArrowProps } from 'react-slick';
+import 'slick-carousel/slick/slick.css';
+import 'slick-carousel/slick/slick-theme.css';
+import { ArrowLeftIcon, ArrowRightIcon } from '@heroicons/react/16/solid';
+
+export type SlideshowProps = {
+  children: React.ReactNode;
```
Comment (rebeccahongsf) [inline src/components/elements/slideshow.tsx]: Change to `React.JSX.Element`

File: src/components/elements/slideshow.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+type SliderRef = Slider | null;
+
+const NextArrow = ({ onClick }: CustomArrowProps) => {
```
Comment (rebeccahongsf) [inline src/components/elements/slideshow.tsx]: Add accessibility (e.g. `aria-label`) to arrows

File: src/components/elements/slideshow.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+const PrevArrow = ({ onClick }: CustomArrowProps) => {
+  return (
+    <button className="absolute top-1/3 left-1 z-50" onClick={onClick}>
+      <ArrowLeftIcon className="w-50 text-black-true bg-teal rounded-full p-10" />
+    </button>
+  );
+};
+
+const Slideshow = ({ children }: SlideshowProps) => {
```
Comment (rebeccahongsf) [inline src/components/elements/slideshow.tsx]: Add `slideshow props` and `props`

File: src/components/elements/slideshow.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
+'use client';
+import React, { useRef } from 'react';
+import Slider, { CustomArrowProps } from 'react-slick';
+import 'slick-carousel/slick/slick.css';
+import 'slick-carousel/slick/slick-theme.css';
```
Comment (rebeccahongsf) [inline src/components/elements/slideshow.tsx]: Remove slick-carousel styles

File: src/components/elements/slideshow.tsx (hunk 4 above)
Comment (pookmish) [inline src/components/elements/slideshow.tsx]: Add slideshow settings and other `...props` that could be applied to the wrapper element.

File: src/components/elements/slideshow.tsx (hunk 1 above)
Comment (rebeccahongsf) [reply src/components/elements/slideshow.tsx:71]: If possible, put together Storybook stories

File: src/components/elements/slideshow.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+const PrevArrow = ({ onClick }: CustomArrowProps) => {
+  return (
+    <button className="absolute top-1/3 left-1 z-50" onClick={onClick}>
+      <ArrowLeftIcon className="w-50 text-black-true bg-teal rounded-full p-10" />
+    </button>
+  );
+};
+
+const Slideshow = ({ children }: SlideshowProps) => {
+  const arrowRef = useRef<SliderRef | null>(null);
```
Comment (pookmish) [inline src/components/elements/slideshow.tsx]: Can you just put `Slider` here and delete the SliderRef Type? 
```suggestion
  const arrowRef = useRef<Slider>(null);
```

File: src/components/elements/slideshow.tsx (hunk 2 above)
Comment (pookmish) [reply src/components/elements/slideshow.tsx]: One thing I normally do is `type Props = HTMLAttributes<HTMLDivElement> & {}` That inherits the `children` prop.

Comment (pookmish) [reply src/components/elements/slideshow.tsx]: That would also satisfy my other comment on the actual component.

File: src/components/elements/slideshow.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      <ArrowLeftIcon className='w-50 text-black-true bg-teal rounded-full p-10' />
+    </button>
+  );
+};
+
+interface SlideshowProps extends HTMLAttributes<HTMLDivElement> {
+  slideshopProps?: Settings;
+}
+
+const Slideshow = ({ children, slideshopProps, ...props }: SlideshowProps) => {
```
Comment (rebeccahongsf) [inline src/components/elements/slideshow.tsx]: `slideshowProps`

**Review by pookmish — APPROVED** [review]
Let's see how far this gets us :smile:

**Pull request merged**